lib/ErrorManager.php

```diff
@@ -19,12 +20,16 @@ public function __construct(Configuration $configuration)
         $this->configuration = $configuration;
     }
 
-    public function error(string $message): void
+    public function error(string $message, ?Throwable $throwable = null): void
     {
         $this->errors[] = $message;
 
         if ($this->configuration->isAbortOnError()) {
-            throw new Exception($message);
+            throw new Exception($message, 0, $throwable);
+        }
+
+        if ($this->configuration->isSilentOnError()) {
+            return;
```

Comment:
Hmm. I see what you're saying... but they do control two different mechanisms. For example, you could want to avoid throwing errors but you DO want the errors to be "echo'd". That doesn't fit into either the `SILENT` or `THROW` mode.

Buuuuut, you did make me realize that I have the logic a bit backwards in `ErrorManager`. I've just updated it so that if `silentOnError === false`, it WILL echo. Previously, I had the `echo` code AFTER the `throw`... so it would throw before it echo'ed.

That is a slight behavior change: if you WERE aborting on error previously, that will still happen, but now you will also see an echo first), but it makes these two mechanisms works correctly and independently.